iceoryx_posh/include/iceoryx_posh/internal/popo/base_subscriber.inl

```diff
@@ -108,6 +108,27 @@ BaseSubscriber<T, Subscriber, port_t>::take() noexcept
     }
 }
 
+template <typename T, typename Subscriber, typename port_t>
+inline cxx::expected<const mepoo::ChunkHeader*, ChunkReceiveResult>
+BaseSubscriber<T, Subscriber, port_t>::takeChunk() noexcept
+{
+    auto result = m_port.tryGetChunk();
+    if (result.has_error())
+    {
+        return cxx::error<ChunkReceiveResult>(result.get_error());
+    }
+    else
+    {
+        auto maybeHeader = result.value();
+        if (maybeHeader.has_value())
+        {
+            return cxx::success<const mepoo::ChunkHeader*>(maybeHeader.value());
+        }
+    }
+    // could move this to a tryGetChunk but then we should remove expected<optional<>> there in the call chain
```

Comment:
would this be a runtime benefit @elBoberido ?

Reply:
could be a minimal runtime benefit. But even if it wouldn't, it still makes sense to also change the `tryGetChunk` method since the nested optional is not used further down in the code

Reply:
If I do this now it will impact several tests. Maybe in the follow-up?

Reply:
I'll mark it as some improvement but will take no action here.